Sources/Utility/PkgConfig.swift

```diff
@@ -18,15 +18,69 @@ public enum PkgConfigError: Swift.Error {
     case nonWhitelistedFlags(String)
 }
 
-/// Get search paths from pkg-config itself.
-///
-/// This is needed because on Linux machines, the search paths can be different
-/// from the standard locations that we are currently searching.
-private let pkgConfigSearchPaths: [AbsolutePath] = {
-    let searchPaths = try? Process.checkNonZeroExit(
-        args: "pkg-config", "--variable", "pc_path", "pkg-config").chomp()
-    return searchPaths?.split(separator: ":").map({ AbsolutePath(String($0)) }) ?? []
-}()
+public struct PkgConfigExecutionDiagnostic: DiagnosticData {
+    public static let id = DiagnosticID(
+        type: AnyDiagnostic.self,
+        name: "org.swift.diags.pkg-config-execution",
+        defaultBehavior: .warning,
+        description: {
+            $0 <<< "Problem executing pkg-config"
+        }
+    )
+}
+
+struct PCFileFinder {
+    /// DiagnosticsEngine to emit warnings
+    let diagnostics: DiagnosticsEngine
+
+    /// Cached results of locations `pkg-config` will search for `.pc` files
+    let pkgConfigPaths: [AbsolutePath]
+
+    /// The built-in search path list.
+    ///
+    /// By default, this is combined with the search paths inferred from
+    /// `pkg-config` itself.
+    static let searchPaths = [
+        AbsolutePath("/usr/local/lib/pkgconfig"),
+        AbsolutePath("/usr/local/share/pkgconfig"),
+        AbsolutePath("/usr/lib/pkgconfig"),
+        AbsolutePath("/usr/share/pkgconfig"),
+    ]
+
+    /// Get search paths from `pkg-config` itself to locate `.pc` files.
+    ///
+    /// This is needed because on Linux machines, the search paths can be different
+    /// from the standard locations that we are currently searching.
+    public init (diagnostics: DiagnosticsEngine
+) {
+        self.diagnostics = diagnostics
+        do {
+            let searchPaths = try Process.checkNonZeroExit(
+            args: "pkg-config", "--variable", "pc_path", "pkg-config").chomp()
+            pkgConfigPaths = searchPaths.split(separator: ":").map({ AbsolutePath(String($0)) })
+        } catch {
+            diagnostics.emit(data: PkgConfigExecutionDiagnostic())
+            pkgConfigPaths = []
+        }
```

Comment:
We should store the result of this execution in a static variable to avoid shelling out for each pkg-config file in the package graph.